src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -632,18 +631,19 @@ internal bool TryRead(out ReadResult result)
             }
         }
 
-        private static void TrySchedule(PipeScheduler scheduler, Action<object> action, object state)
+        private static void ScheduleCallbacks(PipeScheduler scheduler, PipeCompletionCallbacks completionCallbacks)
```

Comment:
Should this be inlined.

Reply:
Actually. why have this method at all? Why not just call UnsafeSchedule direct in the 2 place this is used.

Reply:
Is inlined as is tiny function `[below ALWAYS_INLINE size]`
```
Inlines into 0600003C Pipe:CompleteWriter(ref):this
  [1 IL=0013 TR=000014 06001A8F] [profitable inline] Monitor:Enter(ref,byref)
    [0 IL=0004 TR=000114 06001A90] [FAILED: does not return] Monitor:ThrowLockTakenException()
    [0 IL=0011 TR=000109 06001A91] [FAILED: noinline per IL/cached result] Monitor:ReliableEnter(ref,byref)
  [0 IL=0019 TR=000019 06000039] [FAILED: noinline per IL/cached result] Pipe:CommitUnsynchronized():bool:this
  [0 IL=0032 TR=000025 0600006B] [FAILED: unprofitable inline] PipeCompletion:TryComplete(ref):ref:this
  [2 IL=0046 TR=000037 06000061] [aggressive inline attribute] PipeAwaitable:Complete(byref):this
    [3 IL=0002 TR=000122 06000062] [aggressive inline attribute] PipeAwaitable:ExtractCompletion(byref):this
      [4 IL=0076 TR=000201 06000026] [profitable inline] CompletionData:.ctor(ref,ref,ref,ref):this
  [5 IL=0057 TR=000043 06000069] [below ALWAYS_INLINE size] PipeCompletion:get_IsCompleted():bool:this
  [0 IL=0070 TR=000095 06001A92] [FAILED: noinline per IL/cached result] Monitor:Exit(ref)
  [0 IL=0080 TR=000082 0600004D] [FAILED: has exception handling] Pipe:CompletePipe():this
**[6 IL=0095 TR=000076 06000048] [below ALWAYS_INLINE size] Pipe:ScheduleCallbacks(ref,ref)
    [0 IL=0007 TR=000304 06000094] [FAILED: target not direct] PipeScheduler:UnsafeSchedule(ref,ref):this
  [7 IL=0108 TR=000066 06000049] [aggressive inline attribute] Pipe:TrySchedule(ref,byref)
    [8 IL=0001 TR=000315 06000022] [below ALWAYS_INLINE size] CompletionData:get_Completion():ref:this
    [9 IL=0012 TR=000329 06000025] [below ALWAYS_INLINE size] CompletionData:get_SynchronizationContext():ref:this
    [10 IL=0020 TR=000347 06000024] [below ALWAYS_INLINE size] CompletionData:get_ExecutionContext():ref:this
    [11 IL=0030 TR=000359 06000023] [below ALWAYS_INLINE size] CompletionData:get_CompletionState():ref:this
    [0 IL=0035 TR=000362 06000094] [FAILED: target not direct] PipeScheduler:UnsafeSchedule(ref,ref):this
    [0 IL=0043 TR=000340 0600004A] [FAILED: noinline per IL/cached result] Pipe:ScheduleWithContext(ref,byref)
```
> Why not just call UnsafeSchedule direct in the 2 place this is used.

Called in 4 places? 

![image](https://user-images.githubusercontent.com/1142958/50049880-6fd1a880-00e6-11e9-95f7-34c03553ac72.png)

> Actually. why have this method at all? 

Strongly types the `s_invokeCompletionCallbacks` action to `PipeCompletionCallbacks` rather than `object`

Reply:
> Strongly types the s_invokeCompletionCallbacks action to PipeCompletionCallbacks rather than object

What do you mean? The scheduler is ultimately typed as (`Action<object>, object state`)